torch/csrc/api/src/data/init.cpp

```diff
@@ -0,0 +1,479 @@
+          "Resets any internal state associate with this reader");
+  py::class_<ChunkDataReader<int32_t>, PyChunkDataReader<int32_t>>(
+      data,
+      "ChunkDataReaderInt32T",
+      "Chunk reader performs data chunking and reading of entire chunks with int32_t data")
+      .def(
+          "read_chunk",
+          &ChunkDataReader<int32_t>::read_chunk,
+          "Read an entire chunk",
+          py::arg("chunk_index"),
+          py::return_value_policy::take_ownership)
+      .def(
+          "chunk_count",
+          &ChunkDataReader<int32_t>::chunk_count,
+          "Returns the number of chunks available in this reader")
+      .def(
+          "reset",
+          &ChunkDataReader<int32_t>::reset,
+          "Resets any internal state associate with this reader");
+  py::class_<ChunkDataReader<int64_t>, PyChunkDataReader<int64_t>>(
+      data,
+      "ChunkDataReaderInt64T",
+      "Chunk reader performs data chunking and reading of entire chunks with int64_t data")
+      .def(
+          "read_chunk",
+          &ChunkDataReader<int64_t>::read_chunk,
+          "Read an entire chunk",
+          py::arg("chunk_index"),
+          py::return_value_policy::take_ownership)
+      .def(
+          "chunk_count",
+          &ChunkDataReader<int64_t>::chunk_count,
+          "Returns the number of chunks available in this reader")
+      .def(
+          "reset",
+          &ChunkDataReader<int64_t>::reset,
+          "Resets any internal state associate with this reader");
+  py::class_<ChunkDataReader<float>, PyChunkDataReader<float>>(
+      data,
+      "ChunkDataReaderFloat",
+      "Chunk reader performs data chunking and reading of entire chunks with float data")
+      .def(
+          "read_chunk",
+          &ChunkDataReader<float>::read_chunk,
+          "Read an entire chunk",
+          py::arg("chunk_index"),
+          py::return_value_policy::take_ownership)
+      .def(
+          "chunk_count",
+          &ChunkDataReader<float>::chunk_count,
+          "Returns the number of chunks available in this reader")
+      .def(
+          "reset",
+          &ChunkDataReader<float>::reset,
+          "Resets any internal state associate with this reader");
+  py::class_<ChunkDataReader<double>, PyChunkDataReader<double>>(
+      data,
+      "ChunkDataReaderDouble",
+      "Chunk reader performs data chunking and reading of entire chunks with double data")
+      .def(
+          "read_chunk",
+          &ChunkDataReader<double>::read_chunk,
+          "Read an entire chunk",
+          py::arg("chunk_index"),
+          py::return_value_policy::take_ownership)
+      .def(
+          "chunk_count",
+          &ChunkDataReader<double>::chunk_count,
+          "Returns the number of chunks available in this reader")
+      .def(
+          "reset",
+          &ChunkDataReader<double>::reset,
+          "Resets any internal state associate with this reader");
+}
+
+void init_dataset_bindings_example(PyObject* module) {
```

Comment:
I'd really like us to avoid having to bundle this with our library and loading it by default since this is test-only code. Things like that make our binary unnecessarily large. Can we make this an extension or at least enable conditional loading?

Reply:
Great idea. I have moved this code to the existing cpp extension test and also added unit tests there too!